src/cmd/ksh93/sh/jobs.c

```diff
@@ -374,11 +374,13 @@ int job_reap(register int sig)
 		flags = WUNTRACED|wcontinued;
 	shp->gd->waitevent = 0;
 	oerrno = errno;
+	int was_ttywait_on = sh_isstate(SH_TTYWAIT); /* save tty wait state */
```

Comment:
This codebase usually declares all the variables used at the top of the function body; I'd like to stay consistent with that.